### PR TITLE
Add HEASARC service

### PR DESCRIPTION
**Pull request by cdeil**
@abigailstev @kalgash82 @leyder and I will add HEASARC as a service at the [Python in astronomy 2015 workshop](http://www.lorentzcenter.nl/lc/web/2015/683//info.php3?wsid=683)

http://heasarc.gsfc.nasa.gov/

### CODE REVIEW

**Comment by abigailStev** [comment]
Here's a link for the Batch Browse interface, with an example: https://heasarc.gsfc.nasa.gov/W3Browse/w3batchinfo.html

**Comment by cdeil** [comment]
@leyder, @keflavich  and I hacked on this some more, and the basic example of getting a list of available datasets for a given target / mission works.

There's a ton left to do, but maybe this can be done in future pull requests?
The most important thing would be to have a bundled response and local test, so that this is tested on travis-ci. @keflavich can probably help extract and implement this.

Or does someone want to continue working on this one and make a pull request against this one?

File: astroquery/heasarc/core.py [hunk 1]
```diff
@@ -0,0 +1,216 @@
...
+
+        Parameters
+        ----------
+        object_name : str
+            name of the identifier to query.
+        get_query_payload : bool, optional
+            This should default to False. When set to `True` the method
+            should return the HTTP request parameters as a dict.
+        verbose : bool, optional
+           This should default to `False`, when set to `True` it displays
```
Comment (bsipocz) [inline astroquery/heasarc/core.py]: The indent here is wrong, and I guess you got the sphinx failure because of that.

**Comment by cdeil** [comment]
@abigailstev @kalgash82 @leyder @keflavich I've cleaned this up a bit (removing all the non-implemented methods and tests). If one of you is still working on this and plans to make a PR against this PR, please let me know. If not my suggestion would be to merge this as a bare-bones initial version and then anyone can add to it in new PRs against `master` in the future.

I didn't include a local test because the response is just a plain FITS file and we currently don't do any processing on it, so there's really no use in putting that in a file in the repo.

**Comment by kalgash82** [comment]
@abigailstev @kalgash82 @leyder @keflavich thanks a lot! I can work on it during the next few weeks at some different active levels, but with the idea of having something really useful before the end of May. I don't mind forking your branch and then doing the PR to yours or doing it directly to the astroquery. Whatever is more convenient for you (maybe the second?)

**Comment by keflavich** [comment]
Is there anything we *can* test locally?  For this first implementation, I guess 'no' - that's OK, even though coveralls will whine.

**Comment by kalgash82** [comment]
@keflavich, by locally do you mean with no url connection? using a local database? If really important to check, I could make it happen, I think. At least for testing purposes, nothing else.

**Comment by keflavich** [comment]
@kalgash82 Yes, that's what I mean.   The "problem" right now is that everything is so straightforward that we'd be testing more of the monkeypatching than the actual query code.

**Comment by cdeil** [comment]
@keflavich @kalgash82 Yes, a local test doesn't make sense here for the moment because there's no parsing or processing going on.

Merging this now ... everyone: if you want to continue working on this, please make a new branch off of `astroquery` master and make a new pull request.

**Comment by abigailStev** [comment]
@cdeil Ok sounds good!

**Comment by cdeil** [comment]
And we're live ...
http://astroquery.readthedocs.org/en/latest/heasarc/heasarc.html

:-)

**Comment by kalgash82** [comment]
Awesome!! Thanks @cdeil! I'll be working on giving it more functionality in the next days/weeks unless someone has already done something or thinks that will have more time than me. I will let you know if I need some hand with any trouble :-)

**Comment by abigailStev** [comment]
Thanks @cdeil! This is a really excellent start! :)

**Pull request merged**